api/v1alpha1/aerospikecluster_validating_webhook.go

```diff
@@ -381,34 +384,115 @@ func validateAerospikeConfig(aslog logr.Logger, configSpec *AerospikeConfigSpec,
 	if !ok {
 		return fmt.Errorf("aerospikeConfig.network not a valid map %v", config["network"])
 	}
+	if err := validateNetworkConfig(networkConf); err != nil {
+		return err
+	}
+
+	// namespace conf
+	nsListInterface, ok := config["namespaces"]
+	if !ok {
+		return fmt.Errorf("aerospikeConfig.namespace not a present. aerospikeConfig %v", config)
+	} else if nsListInterface == nil {
+		return fmt.Errorf("aerospikeConfig.namespace cannot be nil")
+	}
+	if nsList, ok := nsListInterface.([]interface{}); !ok {
+		return fmt.Errorf("aerospikeConfig.namespace not valid namespace list %v", nsListInterface)
+	} else if err := validateNamespaceConfig(aslog, nsList, storage, clSize); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateNetworkConfig(networkConf map[string]interface{}) error {
 	if _, ok := networkConf["service"]; !ok {
 		return fmt.Errorf("network.service section not found in config. Looks like object is not mutated by webhook")
 	}
 
+	tlsNames := make(map[string]struct{})
 	// network.tls conf
 	if _, ok := networkConf["tls"]; ok {
 		tlsConfList := networkConf["tls"].([]interface{})
 		for _, tlsConfInt := range tlsConfList {
 			tlsConf := tlsConfInt.(map[string]interface{})
+			if tlsName, ok := tlsConf["name"]; ok {
+				tlsNames[tlsName.(string)] = struct{}{}
+			}
 			if _, ok := tlsConf["ca-path"]; ok {
 				return fmt.Errorf("ca-path not allowed, please use ca-file. tlsConf %v", tlsConf)
 			}
 		}
 	}
-
-	// namespace conf
-	nsListInterface, ok := config["namespaces"]
-	if !ok {
-		return fmt.Errorf("aerospikeConfig.namespace not a present. aerospikeConfig %v", config)
-	} else if nsListInterface == nil {
-		return fmt.Errorf("aerospikeConfig.namespace cannot be nil")
+	if err := validateTlsClientNames(networkConf); err != nil {
+		return err
 	}
-	if nsList, ok := nsListInterface.([]interface{}); !ok {
-		return fmt.Errorf("aerospikeConfig.namespace not valid namespace list %v", nsListInterface)
-	} else if err := validateNamespaceConfig(aslog, nsList, storage, clSize); err != nil {
+	for _, connectionType := range networkConnectionTypes {
+		if err := validateNetworkConnection(networkConf, tlsNames, connectionType); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateTlsClientNames(networkConf map[string]interface{}) error {
```

Comment:
1. tls-authenticate-client can not be "true", That check needs to be added.
2. How are we going to provide the client name of the operator in the tls-authenticate-client list? Are there any more pull reqs coming related to the tls-authenticate-client?

Reply:
Discussed offline:
1. Aerospike can successfully start with tls-authenticate-client=true. If we still want this validation it should be a separate PR with separate discussion.
2. There are many ways how to extract correct name from the cert (different parts of subject name, different extensions in SAN). In order to make it less error prone and make it more flexible we will let user to manually specify names in a config.

Reply:
I believe there's no validation done by aerospike for `true` in `tls-authenticate-client`. We only check for `any` and `false` and anything else would be treated as a tls name and Aerospike will go forward with client authentication enabled.